### PR TITLE
Fix lowercase src highlighting

### LINKED ISSUE #86: Lowercase `#+(begin|end)_src` blocks not syntax highlighted
### Are you using "tree-sitter" branch?

No

### Describe the bug

`#+BEGIN_SRC` and `#+END_SRC` block is properly syntax highlighted, but `#+begin_src` and `#+end_src` block is not.

### Steps to reproduce

This is my test block:
```
#+BEGIN_SRC r
library(tidyverse)
#+END_SRC

#+begin_src r
library(tidyverse)
#+end_src
```

I then saved and reloaded the file.

### Expected behavior

Both blocks should have syntax highlighting for the language specified

### Emacs functionality

Conventionally, uppercase `BEGIN_|END_` is used, but [lowercase is also accepted](https://orgmode.org/manual/Conventions.html)

### OS / Distro

Arch Linux

### Neovim version/commit

v0.6.0-dev+218-g3f526feeb

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by TravonteD**
Closes #86 by adding the lowercase `begin_src/end_src` to the syntax match group for language highlighting

### CODE REVIEW

**Comment by TravonteD** [comment]
@kristijanhusak I'm looking into what's happening with the failing specs. Running the failing test files individually seems to pass, but when running `make test` they fail. This is with the latest `plenary.nvim`. Let me know if you have any insight.

**Comment by kristijanhusak** [comment]
I think I'll have to drop support for versions under 0.7. Nvim-treesitter requires 0.7, and it doesn't have any tag versioning that can be used for older Neovim versions. I'll announce it on the breaking change issue.

**Comment by kristijanhusak** [comment]
Regarding this change, there is an open PR https://github.com/nvim-orgmode/orgmode/pull/89 that converts everything to lowercase, but it's really outdated. Seems like lowercase keywords are the standard https://list.orgmode.org/87tuuw3n15.fsf@nicolasgoaziou.fr/ . I'd like to have everything compatible with the lowercase, not only begin/end src. Is there any specific reason why you set up only begin/end src?

**Comment by TravonteD** [comment]
> Is there any specific reason why you set up only begin/end src?
 
Yes, the issue was that language highlighting didn't work, and that only applies to `src` blocks. `quote` and `example` blocks behave identically already.

**Comment by kristijanhusak** [comment]
@TravonteD do you have any idea why tests return exit status 2? I can't reproduce it locally, it returns 0 correctly.

File: tests/plenary/org/autocompletion_spec.lua [hunk 1]
```diff
@@ -150,8 +150,8 @@ describe('Autocompletion', function()
 
     result = OrgmodeOmniCompletion(0, '#+B')
```
Comment (kristijanhusak) [inline tests/plenary/org/autocompletion_spec.lua:151]:
```suggestion
    result = OrgmodeOmniCompletion(0, '#+b')
```

File: syntax/org_legacy.vim [hunk 2]
```diff
@@ -87,9 +87,9 @@ hi def link org_title           Title
 syntax match  org_verbatim /^\s*>.*/
 syntax match  org_code     /^\s*:.*/
 
-syntax region org_verbatim start="^\s*#+BEGIN_.*"      end="^\s*#+END_.*"      keepend contains=org_block_delimiter
-syntax region org_code     start="^\s*#+BEGIN_SRC"     end="^\s*#+END_SRC"     keepend contains=org_block_delimiter
-syntax region org_code     start="^\s*#+BEGIN_EXAMPLE" end="^\s*#+END_EXAMPLE" keepend contains=org_block_delimiter
+syntax region org_verbatim start="^\s*#+\(BEGIN_\|begin_\).*"      end="^\s*#+\(END_\|end\).*"      keepend contains=org_block_delimiter
```
Comment (kristijanhusak) [inline syntax/org_legacy.vim:90]:
```suggestion
syntax region org_verbatim start="^\s*#+\(BEGIN_\|begin_\).*"      end="^\s*#+\(END_\|end_\).*"      keepend contains=org_block_delimiter
```

File: tests/plenary/org/autocompletion_spec.lua (hunk 1 above)
Comment (kristijanhusak) [reply tests/plenary/org/autocompletion_spec.lua:151]: This is causing tests to fail.

**Comment by kristijanhusak** [comment]
[This line](https://github.com/nvim-orgmode/orgmode/blob/master/lua/orgmode/parser/file.lua#L367) should also be updated to this to properly detect filetypes:
```lua
  local blocks = self:get_ts_matches('(block name: (expr) @name parameter: (expr) @parameters (#any-of? @name "SRC" "src"))')
```

**Review by kristijanhusak — APPROVED** [review]

**Comment by kristijanhusak** [comment]
Thanks!

**Pull request merged**